src/test/java/org/opensearch/sdk/TestSDKNamedWriteableRegistry.java

```diff
@@ -0,0 +1,103 @@
+package org.opensearch.sdk;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opensearch.common.io.stream.NamedWriteable;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.xcontent.*;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+
+public class TestSDKNamedWriteableRegistry extends OpenSearchTestCase {
+    private TestSDKNamedWriteableRegistry.ExampleRunnerForTest runner;
+
+    private static class Example implements NamedWriteable {
+        public static final String NAME = "Example";
+        public static final NamedWriteableRegistry.Entry WRITEABLE_REGISTRY = new NamedWriteableRegistry.Entry(
+                TestSDKNamedWriteableRegistry.Example.class,
+                NAME,
+                null
+        );
+
+        private final String name;
+
+        public Example(String name) {
+            this.name = name;
+        }
+
+
+        @Override
```

Comment:
we probably don't need equals and hashcode for a test class, but it does no harm ...